copycat/file/src/main/java/org/apache/kafka/copycat/file/FileStreamSourceTask.java

```diff
@@ -66,14 +63,14 @@ public void start(Properties props) {
         if (stream == null) {
             try {
                 stream = new FileInputStream(filename);
-                SchemaAndValue offsetWithSchema = context.offsetStorageReader().offset(new SchemaAndValue(OFFSET_KEY_SCHEMA, filename));
-                if (offsetWithSchema != null) {
-                    if (!offsetWithSchema.schema().equals(OFFSET_VALUE_SCHEMA))
-                        throw new CopycatException("Unexpected offset schema.");
-                    Long lastRecordedOffset = (Long) offsetWithSchema.value();
+                Map<String, Object> offset = context.offsetStorageReader().offset(Collections.singletonMap(FILENAME_FIELD, filename));
+                if (offset != null) {
+                    Object lastRecordedOffset = offset.get(POSITION_FIELD);
+                    if (lastRecordedOffset != null && !(lastRecordedOffset instanceof Long))
```

Comment:
I hear ya on offset-reading requiring all sorts of boiler-plate that all connectors will need to suffer from.